birdhouse/components/cowbird/config/cowbird/config.yml.template

```diff
@@ -0,0 +1,155 @@
+        # `/geoserver/workspaces/user_xyz/dir1/dir2/file_abc` that matches with the `geoserver_workspace` resource key,
+        # the `user` variable name would be matched with `user_xyz` and `synced_file`, with `file_abc`.
+        # Also, this key would need to sync permissions with the `thredds_workspace` resource key, considering the
+        # `permissions_mapping` defined below. The `thredds_workspace` would be deduced to the resource path
+        # `/catalog/workspaces/user_xyz/dir1/dir2/subdir/file_abc`.
+        # The types of each segment of this target resource path would be deduced
+        # from the `thredds_workspace` config below.
+        thredds_workspace:
+          - name: catalog
+            type: service
+          - name: workspaces
+            type: directory
+          - name: "{user}"
+            type: directory
+          - name: "**"
+            type: directory
+          - name: subdir
+            type: directory
+          - name: "{synced_file}"
+            type: file
+      geoserver:
+        geoserver_workspace:
+          - name: geoserver
+            type: service
+          - name: workspaces
+            type: workspace
+          - name: "{user}"
+            type: workspace
+          - name: "**"
+            type: workspace
+          - name: "{synced_file}"
+            type: workspace
+    # [Required] This section defines an array of permissions mapping between services
+    # Each item found in the `permissions_mapping` uses the following format :
+    #     resource_key1 : <permissions1> <mapping> resource_key2 : <permissions2>
+    # <permissionsX> is defined as a single permission or a list of permissions :
+    #     `permission | [permission1, permission2, ...]`
+    # <mapping> is defined as a unidirectional or bidirectional arrow : `-> | <- | <->`
+    #
+    # The permissions can either be a single permission or a list of permissions. Each of those permissions can either
+    # use an implicit format (`<name>` or `<name>-match`) or an explicit format (`<name>-<access>-<scope>`).
+    # When using an implicit format, if the access and/or scope are not specified, it will use the default
+    # access `allow` and/or the default scope `recursive`.
+    # For more info on Magpie permissions :
+    #     https://pavics-magpie.readthedocs.io/en/latest/permissions.html#permission-definition-and-modifiers
+    #     https://pavics-magpie.readthedocs.io/en/latest/permissions.html#permissions-representation
+    # See documentation 'Configuration' for more details on the mapping and synchronization features
+    permissions_mapping:
+      # A `read` permission for `thredds_workspace` creates/deletes all specified `geoserver_workspace` permissions.
+      # And a specified `geoserver_workspace` permission creates/deletes a `read` permission on `thredds_workspace`.
+      - "geoserver_workspace : [getCapabilities, getFeature, getFeatureInfo] <-> thredds_workspace : read"
```

Comment:
Indeed. Cowbird does a lot of schema and regex validation of those definitions. 
The one-liner syntax was preserved over nested definitions because they are more readable at a glance, especially when a lot of rules are stacked in a list.